### PR TITLE
Sx11 user can reset password

### PR DESCRIPTION
**Pull request by ahmad3m3**
@daliamaarek: Kindly review the scenario 
@AmirMaurice: Kindly check the diagrams
@ahmad-diaa: Kindly review the code
@maggie-moheb : Kindly check the test
@azamzamy : kindly check the documentation

### CODE REVIEW

File: SmartX/app/src/main/java/com/smartx/cookies/smartx/test/answerSecurityQuestionTest.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import java.util.List;
+
+import models.User;
+import retrofit.Callback;
+import retrofit.RestAdapter;
+import retrofit.RetrofitError;
+import retrofit.client.Response;
+
+/**
+ * Created by ahmad on 30/04/15.
```
Comment (azamzamy) [inline SmartX/app/src/main/java/com/smartx/cookies/smartx/test/answerSecurityQuestionTest.java:23]: @ahmad3m3 add yourself as an author please, in the form of @author

File: SmartX/app/src/main/java/com/smartx/cookies/smartx/test/getQuestionTest.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import com.smartx.cookies.smartx.myAPI;
+
+import models.securityQuestion;
+import retrofit.Callback;
+import retrofit.RestAdapter;
+import retrofit.RetrofitError;
+import retrofit.client.Response;
+
+/**
+ * Created by ahmad on 02/05/15.
```
Comment (azamzamy) [inline SmartX/app/src/main/java/com/smartx/cookies/smartx/test/getQuestionTest.java]: @ahmad3m3 author please

File: SmartX/app/src/main/java/com/smartx/cookies/smartx/myAPI.java [hunk 3]
```diff
@@ -19,6 +20,12 @@
 
 public interface myAPI {
```
Comment (azamzamy) [inline SmartX/app/src/main/java/com/smartx/cookies/smartx/myAPI.java:34]: api methods documentation :eyes: :octocat:

**Comment by ahmad3m3** [comment]
@azamzamy: recheck now please.

**Comment by maggie-moheb** [comment]
bravo r7eemo test is great :+1:

**Comment by AmirMaurice** [comment]
@ahmad3m3 :+1:

**Pull request merged**